onset_maproom/tests/test_calc.py

```diff
@@ -4,10 +4,463 @@
+        0.        ,  0.00165216,  0.        ,  0.        ,  0.        ,
+        0.        ,  0.        ,  0.        ,  0.        ,  0.        ,
+        0.        ,  0.        ,  0.        ,  0.21126945,  5.749138  ,
+        5.4374275 ,  3.3610737 ,  0.00103077,  0.08556359,  0.07880804,
+        2.277596  ,  0.11416379,  0.        ,  0.        ,  0.        ,
+        0.        ,  4.0256405 ,  1.322553  ,  0.        ,  0.2692001 ,
+        0.        ,  0.        ,  1.7372526 ,  0.        ,  0.        ,
+        0.        ,  0.        ,  0.        ,  0.        ,  0.        ,
+        0.        ,  0.        ,  0.        ,  0.        ,  0.        ,
+        0.        ,  0.        ,  0.        ,  0.        ,  0.        ,
+        0.        ,  0.        ,  0.        ,  0.        ,  0.        ,
+        0.        ,  0.45122755,  0.05431353,  0.33383146,  0.6377049 ,
+        6.4067054
+    ]
+    # fmt: on
+    precip = xr.DataArray(values, dims=["T"], coords={"T": t}).rename("precip")
+    return precip
+
+
+def test_daily_tobegroupedby_season_cuts_on_days():
+
+    precip = set_multi_year_data_sample()
+    dts = calc.daily_tobegroupedby_season(precip, 29, 11, 29, 2)
+
+    assert dts["T"].size == 461
+
+
+def test_daily_tobegroupedby_season_creates_groups():
+
+    precip = set_multi_year_data_sample()
+    dts = calc.daily_tobegroupedby_season(precip, 29, 11, 29, 2)
+
+    assert dts["group"].size == 5
+
+
+def test_daily_tobegroupedby_season_picks_right_end_dates():
+
+    precip = set_multi_year_data_sample()
+    dts = calc.daily_tobegroupedby_season(precip, 29, 11, 29, 2)
+    assert (
+        dts.seasons_ends
+        == pd.to_datetime(
+            xr.DataArray(
+                [
+                    "2001-02-28T00:00:00.000000000",
+                    "2002-02-28T00:00:00.000000000",
+                    "2003-02-28T00:00:00.000000000",
+                    "2004-02-29T00:00:00.000000000",
+                    "2005-02-28T00:00:00.000000000",
+                ],
+                dims=["group"],
+                coords={"group": dts.seasons_ends["group"]},
+            )
+        )
+    ).all()
+
+
+def test_seasonal_onset_date_keeps_returning_same_outputs():
+
+    precip = set_multi_year_data_sample()
+    onsetsds = calc.seasonal_onset_date(
+        daily_rain=precip,
+        search_start_day=1,
+        search_start_month=3,
+        search_days=90,
+        wet_thresh=1,
+        wet_spell_length=3,
+        wet_spell_thresh=20,
+        min_wet_days=1,
+        dry_spell_length=7,
+        dry_spell_search=21,
+        time_coord="T",
+    )
+    onsets = onsetsds.onset_delta + onsetsds["T"]
+    assert np.isnat((onsets[0]))
+    assert onsets[1] == pd.to_datetime("2001-03-08T00:00:00.000000000")
+    assert np.isnat((onsets[2]))
+    assert onsets[3] == pd.to_datetime("2003-04-12T00:00:00.000000000")
+    assert onsets[4] == pd.to_datetime("2004-04-04T00:00:00.000000000")
```

Comment:
Why are you making assertions about individual array elements here, rather than comparing the whole array as you did elsewhere? The disadvantage of doing it this way is that the test stops after the first failed assertion, so we don't get the full picture of what works and what doesn't.

Reply:
I see. I found this: https://stackoverflow.com/questions/10710328/comparing-numpy-arrays-containing-nan